npm/private/npm_import.bzl

```diff
@@ -382,7 +382,13 @@ def _download_and_extract_archive(rctx):
             "pattern": "Bearer <password>",
             "password": rctx.attr.npm_auth,
         },
-    } if rctx.attr.npm_auth else {}
+    } if rctx.attr.npm_auth else {
```

Comment:
Can you switch this to:
```
if rctx.attr.npm_auth:
   ...
elif rctx.attr.npm_auth_username and rctx.attr.npm_auth_password:
   ...
else:
   ...
```
Easier to read in that case.

Should the rule `fail()` hard if both `npm_auth` and `npm_auth_username` and `npm_auth_password` are set?

Should the rule `fail()` hard if only one of `npm_auth_username` and `npm_auth_password` is set?